client-runtime/src/main/java/com/microsoft/rest/v2/http/RxNettyAdapter.java

```diff
@@ -42,52 +47,93 @@ private SSLEngine getSSLEngine(String host) {
+            final HttpRequestBody body = request.body();
+            if (body != null) {
+                try (final InputStream bodyStream = body.createInputStream()) {
+                    obsResponse = rxnReq.writeBytesContent(toByteArrayObservable(bodyStream));
+                }
+            }
 
-        if (isSecure) {
-            rxnClient = rxnClient.secure(getSSLEngine(uri.getHost()));
+            result = obsResponse
+                    .map(new Func1<HttpClientResponse<ByteBuf>, HttpResponse>() {
+                        @Override
+                        public HttpResponse call(HttpClientResponse<ByteBuf> rxnRes) {
+                            return new RxNettyResponse(rxnRes);
+                        }
+                    })
+                    .toSingle();
+        } catch (URISyntaxException | IOException e) {
+            result = Single.error(e);
         }
 
-        HttpClientRequest<ByteBuf, ByteBuf> rxnReq = rxnClient
-                .createRequest(HttpMethod.valueOf(request.httpMethod()), uri.toASCIIString())
-                .addHeaders(rxnHeaders);
+        return result;
+    }
+
+    // This InputStream to Observable<byte[]> conversion comes from rxjava-string
+    // (https://github.com/ReactiveX/RxJavaString). We can't just take a dependency on
+    // rxjava-string, however, because they require an older version of rxjava (1.1.1).
+    private static Observable<byte[]> toByteArrayObservable(InputStream inputStream) {
+        return Observable.create(new OnSubscribeInputStream(inputStream, 8 * 1024));
+    }
+
+    private static final class OnSubscribeInputStream extends SyncOnSubscribe<InputStream, byte[]> {
+        private final InputStream is;
+        private final int size;
 
-        Observable<HttpClientResponse<ByteBuf>> obsResponse = rxnReq;
-        if (body != null) {
-            obsResponse = rxnReq.writeStringContent(Observable.just(body));
+        OnSubscribeInputStream(InputStream is, int size) {
+            this.is = is;
+            this.size = size;
         }
 
-        return obsResponse
-                .map(new Func1<HttpClientResponse<ByteBuf>, HttpResponse>() {
-                    @Override
-                    public HttpResponse call(HttpClientResponse<ByteBuf> rxnRes) {
-                        return new RxNettyResponse(rxnRes);
-                    }
-                })
-                .toSingle();
+        @Override
+        protected InputStream generateState() {
+            return this.is;
+        }
+
+        @Override
+        protected InputStream next(InputStream state, Observer<? super byte[]> observer) {
+            byte[] buffer = new byte[size];
+            try {
+                int count = state.read(buffer);
+                if (count == -1)
+                    observer.onCompleted();
+                else if (count < size)
+                    observer.onNext(Arrays.copyOf(buffer, count));
```

Comment:
[Adding this for future reference]: In case we find that we want to optimize by avoiding making additional copy `Arrays.copyOf` then we could use `java.nio.ByteBuffer` instead of `byte[..]` 

```java
byte[] buffer = new byte[size]; ==> java.nio.ByteBuffer buffer = ByteBuffer.allocate(size);
int count = state.read(buffer);  ==> int count = state.read(buffer.array());
Arrays.copyOf(buffer, count)   ==> ByteBuffer.wrap(buffer.array(), 0, count).array();
```